src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -168,16 +170,21 @@ public void ObjectCreationOverloadResolution(ImmutableArray<MethodSymbol> constr
             RefKind returnRefKind = default,
             TypeSymbol returnType = null,
             bool isFunctionPointerResolution = false,
+            bool isExtensionMethodResolution = false,
             in CallingConventionInfo callingConventionInfo = default)
             where TMember : Symbol
         {
             var results = result.ResultsBuilder;
 
+            // No need to check for overridden or hidden methods
+            // if the members were resolved as extension methods.
+            bool checkOverriddenOrHidden = !isExtensionMethodResolution;
```

Comment:
Should we confirm that all containing types are static types derived from `System.Object`?